src/components/App/App.js

```diff
@@ -27,10 +32,32 @@ const GlobalStyle = createGlobalStyle`
   }
 `;
 
-export const App = () => (
-  <>
-    <GlobalStyle />
-    <Header />
-    <Home />
-  </>
-);
+export const App = () => {
+  const [currentView, setCurrentView] = useState(HOME_VIEW);
+  const [currentVideoId, setCurrentVideoId] = useState();
+
+  const viewMoreHandler = (id) => {
+    setCurrentVideoId(id);
+    setCurrentView(DETAIL_VIEW);
+  };
+
+  const goHomeViewHandler = () => {
+    setCurrentView(HOME_VIEW);
+  };
+
+  return (
+    <SearchProvider>
+      <GlobalStyle />
+      <Header />
+      {currentView === HOME_VIEW ? (
+        <Home handleViewMore={viewMoreHandler} />
```

Comment:
If you refresh your SPA you will lose the search or the video you are watching. You can use react-router to fix this problem (: